tests/integration_tests/integration_settings.py

```diff
@@ -0,0 +1,95 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+import os
+
+##################################################################
+# LAUNCH SETTINGS
+##################################################################
+
+# Keep instance (mostly for debugging) when test is finished
+KEEP_INSTANCE = False
+
+# One of:
+#  lxd_container
+#  ec2
+#  gce
+#  oci
+PLATFORM = 'lxd_container'
+
+# The cloud-specific instance type to run. E.g., a1.medium on AWS
+# If the pycloudlib instance provides a default, this can be left None
+INSTANCE_TYPE = None
+
+# Determines the base image to use or generate new images from.
+# Can be the name of the OS if running a stock image,
+# otherwise the id of the image being used if using a custom image
+OS_IMAGE = 'focal'
+
+# Populate if you want to use a pre-launched instance instead of
+# creating a new one. The exact contents will be platform dependent
+EXISTING_INSTANCE_ID = None
+
+##################################################################
+# IMAGE GENERATION SETTINGS
+##################################################################
+
+# Depending on where we are in the development / test / SRU cycle, we'll want
+# different methods of getting the source code to our SUT. Because of
+# this there are a number of different ways to initialize
+# the target environment.
+
+# Can be any of the following:
+# NONE
+#   Don't modify the target environment at all. This will run
+#   cloud-init with whatever code was baked into the image
+# IN_PLACE
+#   LXD/VM only. Mount the source code as-is directly into
+#   the container to override the pre-existing cloud-init code
+# PROPOSED
+#   Install from the Ubuntu proposed repo
+# <ppa repo>, e.g., ppa:cloud-init-dev/proposed
+#   Install from a PPA. It MUST start with 'ppa:'
+# <file path>
+#   A path to a valid package to be uploaded and installed
+# <image id>
+#   A pre-existing (platform dependent) image id
+IMAGE_SOURCE = 'NONE'
+
+##################################################################
+# GCE SPECIFIC SETTINGS
+##################################################################
+# Required for GCE
+GCE_PROJECT = None
+
+# You probably want to override these
+GCE_REGION = 'us-central1'
+GCE_ZONE = 'a'
```

Comment:
`pycloudlib` doesn't do anything with defaulting. You need to pass the values to the library itself through the various constructors. I think we could allow specifying a configuration file to find the setting in. I wouldn't want to automatically use a pre-configured default though. People might have different accounts for different projects, or accidentally typed `pytest` in the root directory without understanding the implications.

Reply:
> pycloudlib doesn't do anything with defaulting. You need to pass the values to the library itself through the various constructors.

We don't pass any parameters in `Ec2Client._get_client`, so _something_ below us must be handling defaulting (in that case, at least). My guess is that it's `boto` itself doing it; GCE's Python library does have similar support IIRC.

This seems like an inconsistency in `pycloudlib`'s interface that we need to iron out: either we should aim to not perform any defaulting, or we should use it wherever available; I think the mixture here is going to lead to confused developers.

> I wouldn't want to automatically use a pre-configured default though.

As mentioned, we already are doing this with EC2. Broadly speaking, I think this is expected behaviour from people, so I don't know if we need to actively avoid it. (Even more broadly speaking, people may even expect this behaviour _because_ it's how boto operates.)

> accidentally typed pytest in the root directory without understanding the implications

Yeah, this would be very easy to trip over even for those of us who do understand the implications and is orthogonal to defaulting: even if I've configured $expensive_cloud explicitly, I wouldn't want to launch instances just by running `pytest`. I wonder if there's anything we can do to exclude them from the "default" collect.

Reply:
Yes, I see the inconsistency now. We can update pycloudlib to use configured defaults (though my personal preference is to make settings explicit). As far as the default collect, it looks like you can put pytest defaults in `tox.ini`. There is an exclude directory option. I'll play around with that.

Reply:
I created a card to capture this